### ISSUE TITLE
https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html does not say or hint at cluster config

### ISSUE THREAD

**Issue by johngrabner** [Lightning-AI/pytorch-lightning#12143]
## 📚 Documentation

I used lightning from the start of my AI journey. Love it.
Even used multiple GPU training, trivial with lightning.

Now I want to train across multiple PCs in my home, I noticed all references to multi-gpu and num_nodes documentation only say to set num_nodes= to some number.  Somehow, there must be a master node setup. Since I relied on lightning to make PyTorch easy, I have not studied native PyTorch ddp. Hopping the lightning documentation would give some info on configuring the environment to allow num_nodes>1.

Here is a good example https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html, but many other places 
including youtube says the same thing, just set num_nodes=2.

I'm not in an institution with millions of dollars that set up a massive cluster, I'm home with 2 PCs each with 2 GPUs.
May I suggest that everywhere you indicate num_nodes>1, you point to some documentation for options for configuring the nodes/cluster so that setting num_node>1 will work?

For typos and doc fixes, please go ahead and:

1. Create an issue.
1. Fix the typo.
1. Submit a PR.

For very simple fixes, you can submit a PR without a linked issue.

Thanks!

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @rohitgr7 @awaelchli @akihironitta

**Comment by johngrabner** [comment]
I just found https://pytorch-lightning.readthedocs.io/en/latest/clouds/cluster.html#slurm-managed-cluster.
I don't want to change the page https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html since I'm
new at num_nodes>1, but if I'm reading this correctly, may I suggest you add 

"setting num_nodes>1 requires a cluster configuration, see https://pytorch-lightning.readthedocs.io/en/latest/clouds/cluster.html#slurm-managed-cluster for details for how to do this.

**Comment by carmocca** [comment]
@johngrabner Would you like to open a PR with these improvements yourself?

**Comment by johngrabner** [comment]
How do I do a PR?
Are you sure my suggestion is sound since I have not tried it yet?

**Comment by awaelchli** [comment]
Hey, for the setup you described (2 PCs in in a network), you should follow [this guide](https://pytorch-lightning.readthedocs.io/en/latest/clouds/cluster.html#general-purpose-cluster). This is for a general purpose (unmanaged) cluster setup.

**Comment by johngrabner** [comment]
Thank you.  Your "this guide" is the same reference I found after opening this ticket, but I used the URL later down the page. I should have used the link for the top of the page like you did.

I will try again to train across multiple PCs after this training run is done.

This ticket is to suggest that the documentation on https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html
be updated to include a link to your reference. Without such a link, a reader like me will be confused not knowing to search for "computing cluster" to find what is assumed to exist for it to be valid to set num_node>1.  After all, I can set gpus=[0,1], and lighting just works. But setting num_nodes=2 requires extra configuration not specified.
